OrbitCore/OrbitFunction.cpp

```diff
@@ -32,24 +17,7 @@ Function::Function(std::string_view name, std::string_view pretty_name,
       size_(size),
       file_(file),
       line_(line) {
-  ResetStats();
-  SetOrbitTypeFromName();
-}
-
-void Function::Select() {
-  LOG("Selected %s at 0x%" PRIx64 " (address_=0x%" PRIx64
-      ", load_bias_= 0x%" PRIx64 ", base_address=0x%" PRIx64 ")",
-      pretty_name_, GetVirtualAddress(), address_, load_bias_,
-      module_base_address_);
-  Capture::GSelectedFunctionsMap[GetVirtualAddress()] = this;
-}
-
-void Function::UnSelect() {
-  Capture::GSelectedFunctionsMap.erase(GetVirtualAddress());
-}
-
-bool Function::IsSelected() const {
-  return Capture::GSelectedFunctionsMap.count(GetVirtualAddress()) > 0;
+  function::Init(this);
```

Comment:
This is a bit weird. If `Init` is only used here maybe just move what `Init` does here and remove `Init`.

Reply:
I moved it to the namespace because when we change this class to proto, we can't do things into constructor, but we can use `function::Init` instead of it.

Reply:
I also had a look a this. And it's a tricky thing. The main problem here is that OrbitFunction has invariants, so it can't be converted into a proto without removing them. ResetStats can probably be avoided by making `stats_` of type FunctionStats (no shared_ptr). For orbit_type_ it gets more complicated because it looks like a caching variable since its value is determined by the function's name. Is there a use case, where orbit_type_ is different from what SetOrbitTypeFromName set? If not, then it can be removed entirely and shouldn't be in a proto since it's redundant information.

Reply:
But actually you're right: it doesn't make sense for now, and when we move to proto it should have different declaration. So, I removed it.

Reply:
@beckerhe I thought about creating function that construct object, set up all the fields and then return it. In this case we can add methods that we need on initialization. About your question "Is there a use case, where orbit_type_ is different from what SetOrbitTypeFromName set?", I don't see that it could be different now, but I guess @pierricgimmig has more context :)

Reply:
> About your question "Is there a use case, where orbit_type_ is different from what SetOrbitTypeFromName set?", I don't see that it could be different now, but I guess @pierricgimmig has more context :)

The type is not necessarily tied to the name of the function, it could eventually be set by the user.  For example, if we wanted to track memory allocations, we'd ask the user to tag the function as "ALLOC".